### PR TITLE
Fix duration of slot conditions

### PR DESCRIPTION
**Pull request by MacChaeger**
- Includes bonus Perish Song test

### CODE REVIEW

**Comment by MacChaeger** [comment]
This is actually not correct...

**Comment by MacChaeger** [comment]
...well it's good enough for 2 player formats, but I'll need to rework it for 4 player formats.

**Comment by MacChaeger** [comment]
Ok now it should work with 4 players as well.

File: sim/battle.ts [hunk 1]
```diff
@@ -802,6 +802,24 @@ export class Battle extends Dex.ModdedDex {
 			});
 			this.resolveLastPriority(handlers, callbackName);
 		}
+		const side = pokemon.side;
+		for (const j in side.slotConditions[pokemon.position]) {
```
Comment (scheibo) [inline sim/battle.ts]: `j` doesn't seem like a good name for this.

File: sim/side.ts [hunk 2]
```diff
@@ -272,6 +272,7 @@ export class Side {
 		const slotConditionData = this.slotConditions[target][status.id] = {
 			id: status.id,
 			target: this,
+			targetPosition: target,
```
Comment (scheibo) [inline sim/side.ts]: when is this used?

Comment (MacChaeger) [reply sim/side.ts]: Never, but I thought it might be nice to have? I'll remove it.

Comment (scheibo) [reply sim/side.ts]: You can keep it if you'd like, I just wanted to make sure I wasn't missing something :)

Comment (MacChaeger) [reply sim/side.ts]: Someone can re-add it in the future if for some reason it becomes necessary, but if it does then something is probably wrong with how slot conditions are implemented.

**Pull request merged**